fs/fs.mbt

```diff
@@ -0,0 +1,159 @@
+// Copyright 2024 International Digital Economy Academy
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub type! IOError {
+  NotFound(String)
+} derive(Show)
+
+/// Writes a string to a file.
+///
+/// # Parameters
+/// - `path`: A `String` representing the file path.
+/// - `content`: A `String` containing the content to be written to the file.
+pub fn write_string_to_file(~path : String, ~content : String) -> Unit {
+  @ffi.write_string_to_file(path, content)
+}
+
+/// Writes an array of bytes to a file at the specified path.
+///
+/// # Parameters
+///
+/// - `path` : The path to the file where the bytes will be written.
+/// - `content` : An array of bytes to be written to the file.
+pub fn write_bytes_to_file(~path : String, ~content : Bytes) -> Unit {
+  @ffi.write_bytes_to_file(path, content)
+}
+
+/// Checks if a path exists.
+///
+/// # Parameters
+/// - `path`: A `String` representing the file path.
+///
+/// # Returns
+/// A boolean indicating whether the path exists.
+pub fn path_exists(~path : String) -> Bool {
+  @ffi.path_exists(path)
+}
+
+/// Reads the entire contents of a file into a string.
+///
+/// # Parameters
+/// - `path`: A `String` representing the file path.
+///
+/// # Returns
+/// A `String` containing the file contents if the file exists, otherwise raises an error.
+pub fn read_file_to_string(~path : String) -> String! {
+  guard path_exists(~path) else {
+    raise IOError::NotFound("file `\{path}` does not exist")
```

Comment:
`raise IOError::NotFound(path)` is enough here. We can customize the `Show` trait for `IOError` to reduce redundant error messages.